src/utils.js

```diff
@@ -18,6 +18,22 @@ genotet.utils.PIXEL_TOLERANCE = .1;
 /** @const {number} */
 genotet.utils.RANGE_TOLERANCE = .001;
 
+/** @const {RegExp} */
+genotet.utils.VALIDATE_EMAIL_REGEX =
+  /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+
+/** @const {RegExp} */
+genotet.utils.VALIDATE_USERNAME_REGEX = /^\w+$/;
```

Comment:
They are not case sensitive. Why also lowercase them on the server? I don't think it is necessary.